pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
```

Comment:
This separation in the API allows us to keep some volumes from users (assuming the PV API is restricted to admins) and the implementation in #5398 was pretty easy in retrospect -- so long as it's TA'd (Thockin Approved)<sup>&trade;</sup>